pageserver/src/repository/inmemory/relfile.rs

```diff
@@ -0,0 +1,436 @@
+///
+/// `relfile` manages storage, caching, and page versioning on a single relation file.
+///
+
+//
+// Currently, the "file format" consists of two parts: base snapshots, and the WAL.
+// A base snapshot consists of a copy of the PostgreSQL data directory, in the normal
+// PostgreSQL file format. A base snapshot is taken at one particular instant, at
+// a specific LSN. Each snapshot is stored in a directory, named like this:
+//
+//    .zenith/timelines/<timelineid>/snapshots/<LSN>/
+//
+// The second component is the WAL, stored in the regular PostgreSQL format, in 16 MB
+// (by default) files. It is stored in:
+//
+//    .zenith/timelines/<timelineid>/wal/
+//
+// At Page Server startup, we load all the WAL into memory, see
+// 'InMemoryRepository::get_or_restore_timeline`. When a get_page_at_lsn() request
+// comes in, we gather all the WAL that applies to the block, read the base image
+// of the page from the latest snapshot on disk, and apply the WAL records on top
+// of the image to reconstruct the requested page version.
```

Comment:
Make all of this `//!`?

Module documentation is often quite verbose, so long docs like this are quite common.